datacompy/core.py

```diff
@@ -156,7 +161,16 @@ def df1(self) -> pd.DataFrame:
 
     @df1.setter
     def df1(self, df1: pd.DataFrame) -> None:
-        """Check that it is a dataframe and has the join columns."""
+        """First, hash any sensitive columns
+        Then, that it is a dataframe and has the join columns."""
+        if self.sensitive_data_columns_df1:
+            for column in self.sensitive_data_columns_df1:
+                if column in df1.columns:
+                    df1[column] = df1[column].apply(lambda v: hashlib.sha256(str(v).encode('utf-8')).hexdigest())
```

Comment:
Can we do some analysis on how fast this operation is? I just want to make sure this doesn't turn out to be some bottleneck in terms of performance if there is a lot of data someone ends up hashing. A pure vectorized approach would be the best.